Client/U2FExtensions.swift

```diff
@@ -311,10 +311,10 @@ class U2FExtensions: NSObject {
                 }
                 
                 return credentialDescriptor
-            })
+            }) ?? []
             
-            guard exclusionList.count == publicKey.excludeCredentials.count else {
-                sendFIDO2AuthenticationError(handle: handle)
+            guard (exclusionList.count == 0 && publicKey.excludeCredentials?.count == nil) || (exclusionList.count == publicKey.excludeCredentials?.count) else {
```

Comment:
Hmm this part: `exclusionList.count == 0  && publicKey.excludeCredentials?.count == nil` I think shouldn't be there.
It basically says: 
`if exclusionList.count != 0 || publicKey.excludeCredentials  { sendFIDO2RegistrationError }`

It's optional so we shouldn't have to guard against it being empty. We only need to guarantee that if we receive exclusionList, it maps perfectly which is what `exclusionList.count == publicKey.excludeCredentials?.count` does :)

Reply:
@Brandon-T - `exclusionList.count == publicKey.excludeCredentials?.count` if `publicKey.excludeCredentials` is not present `publicKey.excludeCredentials?.count` is set to `nil` but `exclusionList.count` is `0`.

Reply:
@jumde I see what you mean. But in that case, you can do:

`guard exclusionList.count == publicKey.excludeCredentials?.count ?? 0 else {...}` instead.

Reply:
Sounds good, thanks!